### PR TITLE
Minor optimizations

### PR DESCRIPTION
**Pull request by cpuguy83**
While profiling dockerd, I noticed the following:

1. When swarm-mode is enabled, the swarm agent is hitting the daemon `info` backend a lot to get information about the system, leading to lots of calls to `imageStore.Map()`
2. The containerd monitor probing was consuming more CPU than expected

On `1`, the `info` backend only actually needs the count of images, but the image store iterates through each image and constructs a new image object. This means the more images there are the more resources are consumed by calling `info`. Instead this just adds an interface to get the image count. Upon doing this, this function drops off the CPU profile (was top of the list before).

On `2`, it's using a ticker to provide a sleep for probing contianerd, but this can actually be ticking faster than the probe actually takes, thus providing no real use. Instead I've made this use a sleep to ensure there is at least some time between probes.

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

File: libcontainerd/remote_daemon.go [hunk 1]
```diff
@@ -263,11 +263,9 @@ func (r *remote) startContainerd() error {
 func (r *remote) monitorConnection(monitor *containerd.Client) {
 	var transientFailureCount = 0
 
-	ticker := time.NewTicker(500 * time.Millisecond)
-	defer ticker.Stop()
-
 	for {
-		<-ticker.C
+		time.Sleep(500 * time.Millisecond)
```
Comment (tonistiigi) [inline libcontainerd/remote_daemon.go]: Shouldn't this do a select with `shutdownContext.Done()` instead?

Comment (tonistiigi) [reply libcontainerd/remote_daemon.go]: Not really related to this fix though. In that case, you would still replace the ticker with `time.After`.

Comment (cpuguy83) [reply libcontainerd/remote_daemon.go]: Updated

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**